scripts/deploy-monitoring.sh

```
@@ -0,0 +1,315 @@
+
+# Setup authentication
+setup_auth() {
+  log_info "Setting up GCP authentication..."
+  
+  # Check for base64-encoded key
+  if [[ -n "${GCP_SA_KEY_BASE64:-}" ]]; then
+    log_info "Using GCP_SA_KEY_BASE64 for authentication"
+    TEMP_KEY_FILE=$(mktemp /tmp/gcp-key-XXXXXX.json)
+    echo "$GCP_SA_KEY_BASE64" | base64 --decode > "$TEMP_KEY_FILE"
+    export GOOGLE_APPLICATION_CREDENTIALS="$TEMP_KEY_FILE"
+    trap "rm -f $TEMP_KEY_FILE" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)